install-and-set-up-conda.sh

```diff
@@ -11,58 +11,64 @@ set -e
 # - Sets up local channel to have highest priority (unless $BIOCONDA_DISABLE_BUILD_PREP=1)
 
 # Extract the versions we should be using from common.sh
-curl -L "https://raw.githubusercontent.com/bioconda/bioconda-common/master/common.sh" > common.sh
+COMMON_GIT_REF=${COMMON_GIT_REF:-master}
+curl -L "https://raw.githubusercontent.com/bioconda/bioconda-common/${COMMON_GIT_REF}/common.sh" > common.sh
+cat common.sh
+
 BIOCONDA_UTILS_TAG=$(grep "^BIOCONDA_UTILS_TAG=" common.sh | cut -f2 -d "=" | sed "s/^v//g")
-MINICONDA_VER=$(grep "^MINICONDA_VER=" common.sh | cut -f2 -d "=")
-MINICONDA_INSTALLATION_DIR="/opt/miniconda"
+MAMBAFORGE_VER=$(grep "^MAMBAFORGE_VER=" common.sh | cut -f2 -d "=")
+MAMBAFORGE_INSTALLATION_DIR="/opt/mambaforge"
 ARCH=$(uname -m)
+MAMBAFORGE_URL="https://github.com/conda-forge/miniforge/releases/download/${MAMBAFORGE_VER}/Mambaforge-${MAMBAFORGE_VER}-Linux-x86_64.sh"
 
 if [[ $(uname) == "Darwin" ]]; then
     OS="MacOSX"
-
+    
     # Remove existing installation on macOS runners
     sudo rm -rf /usr/local/miniconda
-    sudo rm -rf ${MINICONDA_INSTALLATION_DIR}
-    sudo mkdir -p $(dirname $MINICONDA_INSTALLATION_DIR)
-    sudo chown -R $USER $(dirname $MINICONDA_INSTALLATION_DIR)
-
+    sudo rm -rf ${MAMBAFORGE_INSTALLATION_DIR}
+    sudo mkdir -p $(dirname $MAMBAFORGE_INSTALLATION_DIR)
+    sudo chown -R $USER $(dirname $MAMBAFORGE_INSTALLATION_DIR)
+    
     # conda-forge-ci-setup does some additional setup for Mac.
     # Installing bioconda-utils and conda-forge-ci-setup with conda causes dependency conflicts.
     # Installing bioconda-utils and conda-forge-ci-setup with mamba works fine.
     BIOCONDA_ADDITIONAL_INSTALL_PKGS="conda-forge-ci-setup"
 else
-    mkdir -p $(dirname $MINICONDA_INSTALLATION_DIR)
+    mkdir -p $(dirname $MAMBAFORGE_INSTALLATION_DIR)
     OS="Linux"
     BIOCONDA_ADDITIONAL_INSTALL_PKGS=""
 fi
 
 
 # Install miniconda
-curl -L "https://repo.anaconda.com/miniconda/Miniconda3-${MINICONDA_VER}-${OS}-${ARCH}.sh" > miniconda.sh
-bash miniconda.sh -b -p "${MINICONDA_INSTALLATION_DIR}"
+echo Download ${MAMBAFORGE_URL}
+curl -L ${MAMBAFORGE_URL} > mambaforge.sh
+head mambaforge.sh
+bash mambaforge.sh -b -p "${MAMBAFORGE_INSTALLATION_DIR}"
 
-export PATH="${MINICONDA_INSTALLATION_DIR}/bin:${PATH}"
+export PATH="${MAMBAFORGE_INSTALLATION_DIR}/bin:${PATH}"
 
 # Set up channels
 conda config --set always_yes yes
 conda config --system --add channels defaults
 conda config --system --add channels bioconda
 conda config --system --add channels conda-forge
 conda config --system --set channel_priority strict
-conda info
+mamba info
 
-conda install mamba -y
+mamba install mamba -y
```

Comment:
Should we omit this line so the mamba version is pinned to what comes with mambaforge?

Reply:
I think it does not hurt to update it. Just to have the latest bug fixes and avoid a situation where mambaforge is behind a recent fix.

Reply:
Performance wise, this does only take a few seconds.